pkg/framework/test/apiserver.go

```diff
@@ -34,17 +35,28 @@ type certDirManager interface {
 var apiServerBinPathFinder = DefaultBinPathFinder
 
 // NewAPIServer creates a new APIServer Fixture Process
-func NewAPIServer(config *APIServerConfig) *APIServer {
+func NewAPIServer(config *APIServerConfig) (*APIServer, error) {
 	starter := func(command *exec.Cmd, out, err io.Writer) (SimpleSession, error) {
 		return gexec.Start(command, out, err)
 	}
 
+	etcd, err := NewEtcd()
```

Comment:
Can we take this as a parameter instead? Or put that as part of the config and default to NewEtcd if unset?

Reply:
We're aiming to make these constructors take no params, so you can use them if you want sensible defaults. If you want to provide your own Etcd, you can either use `APIServer{Etcd: myEtcd}` or `myAPIServer := NewAPIServer() ; myAPIServer.Etcd = myEtcd`.

We're still part-way through removing the parameters of the APIServer constructor, but that's where we were aiming.